src/Microsoft.AspNet.JsonPatch/Helpers/PropertyHelpers.cs

```diff
@@ -62,17 +23,29 @@ public static PropertyInfo FindProperty(object targetObject, string propertyPath
                 // skip the first one if it's empty
                 var startIndex = (string.IsNullOrWhiteSpace(splitPath[0]) ? 1 : 0);
 
-                for (int i = startIndex; i < splitPath.Length - 1; i++)
+                for (int i = startIndex; i <= splitPath.Length - 1; i++)
                 {
-                    var propertyInfoToGet = GetPropertyInfo(targetObject, splitPath[i]
-                        , BindingFlags.IgnoreCase | BindingFlags.Public | BindingFlags.Instance);
-                    targetObject = propertyInfoToGet.GetValue(targetObject, null);
+                    var jsonContract = (JsonObjectContract)contractResolver.ResolveContract(targetObject.GetType());
```

Comment:
Code in lines 45 - 50. I am checking if `PropertyType` is of type `IList`; if yes then getting the object at index position.